### PR TITLE
builder: fix copy —from conflict with force pull

### PR DESCRIPTION
**Pull request by tiborvass**
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>
(cherry picked from commit e430b58811813084df2b9f8b1a9e929114b2187a)
Signed-off-by: Tibor Vass <tibor@docker.com>

### CODE REVIEW

**Comment by tiborvass** [comment]
It's WIP until moby/moby#33735 is merged

File: components/engine/daemon/build.go [hunk 1]
```diff
@@ -104,13 +105,19 @@ func (daemon *Daemon) pullForBuilder(ctx context.Context, name string, authConfi
 // Every call to GetImageAndReleasableLayer MUST call releasableLayer.Release() to prevent
 // leaking of layers.
 func (daemon *Daemon) GetImageAndReleasableLayer(ctx context.Context, refOrID string, opts backend.GetImageAndLayerOptions) (builder.Image, builder.ReleaseableLayer, error) {
-	if !opts.ForcePull {
-		image, _ := daemon.GetImage(refOrID)
+	id, _ := daemon.GetImageID(refOrID)
+	refIsID := id.String() == refOrID // detect if ref is an ID to skip pulling
```
Comment (dnephin) [inline components/engine/daemon/build.go]: Let's not merge this just yet. This really doesn't seem like the right fix to me.

Either we should use the official "is ref or ID" logic from `GetImageID()` or we should have the caller pass in `opts.ForcePull = false` for this case (when it knows it has an ID).

**Comment by tiborvass** [comment]
Closing for now

**Comment by tonistiigi** [comment]
@tiborvass Can you reopen with updated fix

**Comment by thaJeztah** [comment]
ping @tiborvass @tonistiigi what's the status on this one?

**Comment by tiborvass** [comment]
Ping @tonistiigi sorry for the delay.

**Comment by thaJeztah** [comment]
Compile is failing on Windows;

```
05:41:10 # github.com/docker/cli/vendor/github.com/docker/docker/pkg/system
05:41:10 ..\..\vendor\github.com\docker\docker\pkg\system\init.go:10: maxTime redeclared in this block
05:41:10 	previous declaration at ..\..\vendor\github.com\docker\docker\pkg\system\chtimes.go:11
05:41:10 ..\..\vendor\github.com\docker\docker\pkg\system\path_windows.go:13: DefaultPathEnv redeclared in this block
05:41:10 	previous declaration at ..\..\vendor\github.com\docker\docker\pkg\system\path.go:10
```

**Comment by vieux** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

**Pull request merged**